crates/matrix-sdk/src/event_cache/linked_chunk.rs

```diff
@@ -40,7 +100,7 @@ pub enum LinkedChunkError {
 /// instead of a single one. A chunk has a maximum capacity of `CHUNK_CAPACITY`.
 /// Once a chunk is full, a new chunk is created. Not all chunks are necessarily
 /// entirely full. A chunk can represents a `Gap` between other chunks.
-pub struct LinkedChunk<const CHUNK_CAPACITY: usize, Item, Gap> {
+pub struct LinkedChunk<const CHUNK_CAPACITY: usize, Item, Gap, Listener = ()> {
```

Comment:
Does it have to be a generic parameter? Making it a `Arc<dyn Listener>` (optional, or a list of those) in the struct would mean that we could have multiple listeners with different concrete types, which would be nice.

Reply:
`LinkedChunkListener` removed 🙂 .